pkg/cloudprovider/providers/vsphere/nodemanager.go

```diff
@@ -45,10 +45,12 @@ type NodeManager struct {
 	nodeInfoMap map[string]*NodeInfo
 	// Maps node name to node structure
 	registeredNodes map[string]*v1.Node
+	removedNodes    map[string]*v1.Node
```

Comment:
Node information will still stay even if all the volumes that are attached by the cluster are detached. I would suggest to remove this information when not needed.